libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -272,6 +274,183 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
 			manipulator_group.apply_transform(affine_transform);
 		}
 	}
+
+	// Smooths a Subpath up to the first derivative, using a weighted averaged based on segment length.
+	pub(crate) fn smooth_cubic_paths(&mut self) {
+		for i in 0..self.len() - 2 {
```

Comment:
In other parts of this PR, the iterators range from 1 rather than 0, and the indexing happens for `i - 1` and `i` rather than `i` and `i + 1` which is nice because it more clearly avoids OB1 mistakes since it's obvious that `i - 1` when `i` starts at 1 won't go below 0. Would it make sense to adopt that pattern here too, with a `for i in 1..self.len() - 1` loop (and changes to the usage of i below)?

Reply:
Sure, I believe its an easy change.